src/compiler/ng-jest-compiler.spec.ts

```diff
@@ -1,34 +1,587 @@
+
+            expect(diagnostics.length).toBe(0);
+            expect(code).toContain(dedent`
+               MyDir.propDecorators = {
+                 trigger: [{ type: core_1.Input }],
+                 fromOtherFile: [{ type: core_1.Input }]
+               };
+            `);
+            expect(code).not.toContain('HTMLElement');
+            expect(code).not.toContain('MyOtherClass');
+        });
+
+        it.failing('should capture constructor type metadata with `emitDecoratorMetadata` enabled', () => {
+            // This test only passes when doing `import type {MyOtherClass} from './other-file';`
+            const { code, diagnostics } = transformCjs(
+                `
+                    import {Directive} from '@angular/core';
+                    import {MyOtherClass} from './other-file';
+
+                    @Directive()
+                    export class MyDir {
+                        constructor(other: MyOtherClass) {}
+                    }
+                `,
+                { emitDecoratorMetadata: true },
+            );
+
+            expect(diagnostics.length).toBe(0);
+            expect(code).toContain('const other_file_1 = require("./other-file");');
+            expect(code).toContain(dedent`
+                MyDir.ctorParameters = () => [
+                    { type: other_file_1.MyOtherClass }
+                ];
+                exports.MyDir = MyDir = tslib_1.__decorate([
+                    (0, core_1.Directive)(),
+                    tslib_1.__metadata("design:paramtypes", Object])
+                ], MyDir);
+            `);
+        });
+
+        it('should capture constructor type metadata with `emitDecoratorMetadata` disabled', () => {
+            const { code, diagnostics } = transformCjs(
+                `
+                import {Directive} from '@angular/core';
+                import {MyOtherClass} from './other-file';
+
+                @Directive()
+                export class MyDir {
+                    constructor(other: MyOtherClass) {}
+                }
+            `,
+                { emitDecoratorMetadata: false },
+            );
+
+            expect(diagnostics.length).toBe(0);
+            expect(code).toContain('const other_file_1 = require("./other-file");');
+            expect(code).toContain(dedent`
+               MyDir.ctorParameters = () => [
+                    { type: other_file_1.MyOtherClass }
+               ];
+               exports.MyDir = MyDir = tslib_1.__decorate([
+                    (0, core_1.Directive)()
+               ], MyDir);
+            `);
+        });
+
+        it.failing('should properly serialize constructor parameter with external qualified name type', () => {
+            // This test doesn't pass because `transpileModule` can't resolve `import *`
+            const { code, diagnostics } = transformCjs(`
+                import {Directive} from '@angular/core';
+                import * as externalFile from './other-file';
```

Comment:
I'm surprised this would be different than `import { MyOtherClass } from './other-file';` and makes me wonder if there's a bug in [`entityNameToExpression`](https://github.com/angular/angular/blob/5dcdbfcba934a930468aec140a7183b034466bdf/packages/compiler-cli/src/transformers/jit_transforms/downlevel_decorators_transform.ts#L351) somewhere? We have special handling of aliased imports, I wonder if we also need special handling for namespaced imports?

Reply:
I can confirm import name and import default work, only namespace import doesn't